### ISSUE TITLE
Corner case issue with --redundantSelf

### ISSUE THREAD

**Issue by iandundas** [nicklockwood/SwiftFormat#432]
For the following code: 

```swift
protocol SomeType {
    func someOtherFunction() // i.e. not available on SomeClass
}

class SomeClass {
    func someFunction() {
        if let self = self as? SomeType {
            self.someOtherFunction()
        }
    }
}
```

Running `swiftformat test.swift --disable redundantSelf` will produce:

```swift
protocol SomeType {
    func someOtherFunction() // i.e. not available on SomeClass
}

class SomeClass {
    func someFunction() {
        if let self = self as? SomeType {
            someOtherFunction()  // `self` was stripped, no longer compiles: "Use of unresolved identifier 'someOtherFunction'"
        }
    }
}
```

`swiftformat test.swift --disable redundantSelf` does not make this change.

Casting `self` this way is obviously not a great idea code-wise, but it does compile, so the formatter should ideally not output non-compiling code here.

**Comment by nicklockwood** [comment]
It's a corner case, but it should still be handled correctly. I'll look into it - thanks for reporting.

**Comment by nicklockwood** [comment]
Fixed in version 0.40.9.

**Comment by iandundas** [comment]
Thanks!
